### DISCUSSION TITLE
Best way to create a zod validator for a list of literal strings

### DISCUSSION THREAD

**Discussion by SonofNun15** [colinhacks/zod#2890] [Q&A] [answered]
I want to do something like this:

```
export const levels = ['level1', 'level2', 'level3', 'level4'] as const;

export const levelValidator = z.union(
  levels.map(l => z.literal(l))
);

export type Level = z.infer<typeof levelValidator>;
```

Goal is to create a list of literal strings and enforce that the property contains one of these strings.

I could certainly do:
```
z.union([
  z.literal('level1'),
  z.literal('level2'),
  z.literal('level3'),
  z.literal('level4'),
]);
```

But I want to have an array of these values that I can render as the options for a select input.

That map doesn't work because the type definition for `union` is so strict that it enforces that the array has at least 2 elements. This array does, but the compiler can't enforce that.

Any suggestions on the best way to proceed?

This issue is similar:
https://github.com/colinhacks/zod/issues/2691

But I can't imagine using ~100 lines of code for this.

**Comment by JacobWeisenburger** [comment, marked as answer]
https://github.com/colinhacks/zod#zod-enums
